src/shapes/object.class.js

```diff
@@ -1980,25 +1971,19 @@
    * @constant
    * @type string[]
    */
-  fabric.Object.ENLIVEN_PROPS = ['clipPath'];
 
-  fabric.Object._fromObject = function(className, object, callback, extraParam) {
-    var klass = fabric[className];
-    object = clone(object, true);
-    fabric.util.enlivenPatterns([object.fill, object.stroke], function(patterns) {
-      if (typeof patterns[0] !== 'undefined') {
-        object.fill = patterns[0];
-      }
-      if (typeof patterns[1] !== 'undefined') {
-        object.stroke = patterns[1];
-      }
-      fabric.util.enlivenObjectEnlivables(object, object, function () {
-        var instance = extraParam ? new klass(object[extraParam], object) : new klass(object);
-        callback && callback(instance);
-      });
+  fabric.Object._fromObject = function(className, object, extraParam) {
+    var klass = fabric[className], serializedObject = clone(object, true);
```

Comment:
I think we should get rid of `className` and pass a live class (`klass`)

Reply:
yes we can, but that is not fundamental for moving this forward. this is ok as an action item

Reply:
alright